endpoints/events/vtrack.go

```diff
@@ -269,25 +270,34 @@ func getAccountId(httpRequest *http.Request) string {
 	return httpRequest.URL.Query().Get(AccountParameter)
 }
 
-// modifyVastXml modifies BidCacheRequest element Vast XML data
-func modifyVastXml(externalUrl string, data json.RawMessage, bidid string, bidder string, accountId string, timestamp int64) json.RawMessage {
-	c := string(data)
+// modifyVastXmlString rewrites and returns the string vastXML or an empty string if it was left unmodified
+func ModifyVastXmlString(externalUrl string, c string, bidid string, bidder string, accountId string, timestamp int64) string {
 	ci := strings.Index(c, ImpressionCloseTag)
 
 	// no impression tag - pass it as it is
 	if ci == -1 {
-		return data
+		return ""
```

Comment:
Why the change to an empty string from passing it through? aka.. why not continue returning `c`?

Reply:
Empty string indicates no change was needed, which for `ModifyVastXmlJSON` means to skip the conversion back to RawMessage. I felt this was ok given that empty string is not a possible outcome from doing the replacement on any non-empty input.

Reply:
Gotcha. Do you think it would be clearer to add an "ok" boolean return, like Go does for map retrievals?